src/c++/perf_analyzer/data_loader.cc

```diff
@@ -38,6 +38,34 @@ DataLoader::DataLoader(const size_t batch_size)
 {
 }
 
+cb::Error
+DataLoader::ValidateIOExistsInModel(
+    const std::shared_ptr<ModelTensorMap>& inputs,
+    const std::shared_ptr<ModelTensorMap>& outputs,
+    const std::string& data_directory)
+{
+  if (!std::filesystem::exists(data_directory) ||
+      !std::filesystem::is_directory(data_directory)) {
+    return cb::Error(
+        "Error: Directory does not exist or is not a directory: " +
+            std::string(data_directory),
+        pa::GENERIC_ERROR);
+  }
+
+  for (const auto& file : std::filesystem::directory_iterator(data_directory)) {
+    std::string io_name = file.path().filename().string();
+    if (inputs->find(io_name) == inputs->end() &&
+        outputs->find(io_name) == outputs->end()) {
```

Comment:
If a directory is passed to `--input-data`, it is only supposed to hold a binary file for each input:

https://github.com/triton-inference-server/client/blob/main/src/c%2B%2B/perf_analyzer/docs/input_data.md#input-data

We should not be checking against the outputs.

Reply:
Gadzooks!

https://github.com/triton-inference-server/client/blob/50fb4c913770524cfac5059325906d94644d583e/src/c%2B%2B/perf_analyzer/data_loader.cc#L103

Looks like outputs can be read from a directory passed in. The documentation should be updated.

Reply:
I think @tgerdesnv was saying that output validation files should also be considered valid:
https://github.com/triton-inference-server/client/pull/612#discussion_r1581090619.

Reply:
Haha you beat me to it, and I love the saying "gazdooks!". Is the link I cited enough or where would you like it updated?

Reply:
https://github.com/triton-inference-server/client/pull/705